tests/unit_tests/test_example_code.py

```diff
@@ -0,0 +1,53 @@
+import importlib
+import sys
+
+import pytest
+
+from blueapi.core.context import BlueskyContext
+
+sys.path.insert(0, "tests/unit_tests/code_examples")
+
+
+@pytest.mark.parametrize(
+    "module_name,plan_names",
+    [
+        ("plan_module", ["my_plan"]),
+        ("plan_docstrings", ["temp_pressure_snapshot"]),
+        ("plan_metadata", ["pass_metadata"]),
```

Comment:
I believe deferred_plans can/should be added here, as it's otherwise not tested?